### PR TITLE
feat actionHook delay action

### PR DESCRIPTION
**Pull request by xquanluu**
#451

### CODE REVIEW

File: lib/tasks/gather.js [hunk 1]
```diff
@@ -197,6 +197,21 @@ class TaskGather extends SttTask {
       throw new Error('Cobalt requires a model to be specified');
     }
 
+    // actionHook delay
+    this._actionHookDelayEnabled = cs.actionHookDelayEnabled ||
+      (this.actionHookDelayAction !== null && this.actionHookDelayAction !== undefined);
```
Comment (davehorton) [inline lib/tasks/gather.js:202]: is it necessary to test against both null and undefined?  Please research if this gives the same result
```
this._actionHookDelayEnabled = !!cs.actionHookDelayEnabled 
```

Comment (xquanluu) [reply lib/tasks/gather.js]: followed suggestion

**Comment by davehorton** [comment]
@xquanluu can you make this draft state for now until customer has tested?

**Comment by vdharashive** [comment]
@xquanluu 
it did not work enclosed the logs


[hookdelay.txt](https://github.com/jambonz/jambonz-feature-server/files/13827348/hookdelay.txt)

**Comment by xquanluu** [comment]
[
    {
      "verb": "config",
      "synthesizer": {
        "vendor": "microsoft",
        "voice": "en-AU-NatashaNeural",
        "language": "en-AU"
      },
      "recognizer": {
        "vendor": "microsoft",
        "language": "en-AU"
      },
      "actionHookDelayAction": {
        "enabled": true,
        "noResponseTimeout": 4,
        "noResponseGiveUpTimeout": 20,
        "retries": 3,
        "actions": [
          {
            "verb": "say",
            "text": "hi there this is bot no input test",
            "synthesizer": {
              "vendor": "microsoft",
              "language": "en-US"
            }
          },
          {
            "verb": "play",
            "url": "https://file-examples.com/storage/feee3c905665949b89ba53e/2017/11/file_example_WAV_2MG.wav",
            "timeoutSecs": 10,
            "seekOffset": 8000
          }
        ]
      }
    },
    {
      "verb": "tag",
      "data": {
        "userNoInputCounter": 1,
        "retryCount": 3
      }
    },
    {
      "verb": "gather",
      "actionHook": "/actions/hooks",
      "input": [
        "digits",
        "speech"
      ],
      "interDigitTimeout": 2,
      "minDigits": 1,
      "finishOnKey": "#",
      "numDigits": 12,
      "say": {
        "text": "Sample prompt message for Entity0001"
      },
      "bargein": false,
      "listenDuringPrompt": false,
      "timeout": 5,
      "dtmfBargein": false
    }
  ]

**Comment by vdharashive** [comment]
@xquanluu can you merge this PR , we have tested and working as expected

**Comment by vdharashive** [comment]
@davehorton can this be merged

**Comment by davehorton** [comment]
no, see related issue I am waiting for more detail on how this new feature works

**Review by davehorton — APPROVED** [review]

**Pull request merged**